packages/enzyme/src/Utils.js

```diff
@@ -266,3 +266,34 @@ export function cloneElement(adapter, el, props) {
     { ...el.props, ...props },
   );
 }
+
+export function spyMethod(instance, methodName) {
+  let lastReturnValue;
+  const originalMethod = instance[methodName];
+  const hasOwn = Object.hasOwnProperty.call(instance, methodName);
+  Object.defineProperty(instance, methodName, {
+    configurable: true,
+    enumerable: false,
+    value(...args) {
+      const result = originalMethod.apply(this, args);
+      lastReturnValue = result;
+      return result;
+    },
+  });
+  return {
+    restore() {
+      if (hasOwn) {
+        /* eslint-disable no-param-reassign */
+        instance[methodName] = originalMethod;
```

Comment:
this won't actually restore the original property descriptor; to do this properly, it needs to use `Object.getOwnPropertyDescriptor` and `Object.defineProperty` when available, and fall back to normal reading and assignment.